src/tls/openssl/tls.c

```diff
@@ -205,6 +207,19 @@ int tls_verify_handler(int ok, X509_STORE_CTX *ctx)
 #endif
 
 
+static int tls_verify_idx = -1;
+static once_flag oflag = ONCE_FLAG_INIT;
```

Comment:
is it possible to avoid this global data and use the TLS context instead ?

Reply:
I was trying to do that but it is not possible. It is not nice but openssl requires global indices initialized once. in tls_alloc call_once(&oflag, tls_init_verify_idx) is called.

Code examples are taken from openssl and apache2/httpd implementation to get application inside the openssl verify handlers, which are not supporting application data (so no tls context is present). So the only way to get application data is to use global openssl indices initialized once. (SSL_set_ex_data, SSL_get_ex_data(..., tls_verify_idx))